### PR TITLE
[EASI-2605] Front End work for Treat as Other Operational Solution

### PR DESCRIPTION
**Pull request by garyjzhao**
# EASI-2605

## Changes and Description

- this is the same as #542, but with a clean branch off from main

- Deprecate rendering custom solution card on AddCustom view
- `treatAsOther` will be a boolean field on possible AND actual solutions, will not appear on custom solutions
- For this `treatAsOther` solutions outlined [here](https://www.figma.com/file/BcI7pNwfkmwaDD2EGGTsYC/IT-Lead?node-id=1089%3A175128&t=cPbn5sFgs1EBFGHD-0) and [here](https://www.figma.com/file/BcI7pNwfkmwaDD2EGGTsYC/IT-Lead?node-id=415%3A164083&t=cPbn5sFgs1EBFGHD-0) when selecting these from AddSolution view, render the same button below as if you selected 'Other'.  
    - This should take them to the AddCustomSolution view where they can fill out details
- Updated AddCustomSolution view with updated verbiage to account for additional custom types
- Verbiage is dependent on the type of custom solution they selected
- Testing

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: src/views/ModelPlan/TaskList/ITSolutions/AddSolution/index.tsx [hunk 1]
```diff
@@ -322,16 +322,23 @@ const AddSolution = () => {
...
-                            {values.key ===
-                              OperationalSolutionKey.OTHER_NEW_PROCESS &&
+                            {(values.key ===
+                              OperationalSolutionKey.CONTRACTOR ||
+                              values.key ===
+                                OperationalSolutionKey.CROSS_MODEL_CONTRACT ||
+                              values.key ===
+                                OperationalSolutionKey.EXISTING_CMS_DATA_AND_PROCESS ||
+                              values.key ===
+                                OperationalSolutionKey.OTHER_NEW_PROCESS) &&
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ITSolutions/AddSolution/index.tsx:332]: I think there is a solution missing from here.  Also I think we can probably condense this to have an entity containing all the `other` enums and perform `.includes(value.key)`

File: src/views/ModelPlan/TaskList/ITSolutions/_components/CheckboxCard/index.tsx [hunk 2]
```diff
@@ -68,13 +70,31 @@ const CheckboxCard = ({
             checked={!!solution.needed}
           />
 
-          <h3 className="margin-y-2">{solution.nameOther || solution.name}</h3>
+          {solution.key === OperationalSolutionKey.CONTRACTOR ||
+          solution.key === OperationalSolutionKey.CROSS_MODEL_CONTRACT ||
+          solution.key ===
+            OperationalSolutionKey.EXISTING_CMS_DATA_AND_PROCESS ? (
+            <>
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ITSolutions/_components/CheckboxCard/index.tsx:77]: Should this include all the `other` types or just these 3?

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
Nice Gary :) 

This translation does not render correctly.
Screenshot 2023-04-24 at 10 23 42 AM

For solutions that are custom, there should be no `About this solution` link
Screenshot 2023-04-24 at 10 26 39 AM

Selecting "Cross model contract" added two duplicated solutions, with different renderings
Screenshot 2023-04-24 at 10 27 29 AM

There's some funkiness in with the possible needs.  Just want to note I covered this in https://github.com/CMSgov/mint-app/pull/534 that is still in review.
Screenshot 2023-04-24 at 10 29 47 AM

Adding one of the custom solutions does not bring me to the custom solution screen, it automatically adds it as a regular solution.  These solutions should be treated as other

```
  | OperationalSolutionKey.CONTRACTOR
  | OperationalSolutionKey.CROSS_MODEL_CONTRACT
  | OperationalSolutionKey.EXISTING_CMS_DATA_AND_PROCESS
  | OperationalSolutionKey.INTERNAL_STAFF
  | OperationalSolutionKey.OTHER_NEW_PROCESS;
  ```
Screenshot 2023-04-24 at 10 33 52 AM

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
I think this card still needs to read `Add details` and route them to the custom page

Screenshot 2023-04-28 at 10 51 31 AM

https://www.figma.com/file/BcI7pNwfkmwaDD2EGGTsYC/IT-Lead?node-id=1057-169006&t=TpiREdHbk7ZDEfx3-0

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
I got this error app crash when pressing back on the browser from the /add-solution page to the /select-solution page

Screenshot 2023-04-28 at 10 52 26 AM

**Comment by patrickseguraoddball** [comment]
I was also still getting instances of solution that weren't selected showing up randomly on `/solution-implementation-details`

File: src/views/ModelPlan/TaskList/ITSolutions/SolutionImplementation/index.tsx [hunk 3]
```diff
@@ -59,13 +59,14 @@ const SolutionImplementation = ({
...
-    fromSolutionDetails: boolean;
-    isCustomNeed: boolean;
+  const { state } = useLocation<{
+    fromSolutionDetails?: boolean;
+    isCustomNeed?: boolean;
   }>();
 
+  const fromSolutionDetails = state?.fromSolutionDetails;
+  const isCustomNeed = state?.isCustomNeed;
+
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ITSolutions/SolutionImplementation/index.tsx:69]: This destructuring needs to be addressed in `SelectSolutions` as well

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]

File: src/views/ModelPlan/TaskList/ITSolutions/AddCustomSolution/index.tsx [hunk 4]
```diff
@@ -92,6 +106,12 @@ const AddCustomSolution = () => {
     operationalSolutionID?: string;
   }>();
 
+  const {
+    state: { selectedSolution }
+  } = useLocation<{
+    selectedSolution?: SelectedOpertionalSolutionKeyType;
+  }>();
+
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ITSolutions/AddCustomSolution/index.tsx:114]: This shouldn't be destructured here as well

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]

**Comment by patrickseguraoddball** [comment]
I'm also a bit torn on whether to use location state at all to pass state between routes.  I think a more consistent method would be to append query params to the route to ensure that on app refresh/redirect the state is persisted.  If you want to leave this out that might be okay, but we should ticket it as tech debt?

**Comment by garyjzhao** [comment]
Tech debt created to address using query params, instead of location state: https://jiraent.cms.gov/browse/EASI-2898

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
When I added "Internal staff" as a custom solution, then chose to "Add details" to it, it create an entirely new solution instead of adding details to "Internal staff"

Screenshot 2023-05-03 at 2 43 23 PM

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
It's possible for solutions to have identical keys with the ability to re-add the same solution.  React throwing the duplicate key error:

Screenshot 2023-05-03 at 3 24 44 PM

Screenshot 2023-05-03 at 3 24 33 PM

Also getting these new console errors on the Add Custom solution detail page

Screenshot 2023-05-03 at 3 27 57 PM

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
I think the point of contact needs to be hidden if there is no point of contact.  Also the email icon is rendered with no email

Screenshot 2023-05-08 at 11 25 38 AM

Screenshot 2023-05-08 at 11 25 26 AM

File: src/views/ModelPlan/TaskList/ITSolutions/_components/CheckboxCard/index.tsx [hunk 5]
```diff
@@ -84,26 +147,55 @@ const CheckboxCard = ({
...
+              !isDefaultSolutionOptions && (
+                <>
+                  <h3 className="margin-top-2 margin-bottom-0">
+                    {solution.otherHeader}
+                  </h3>
+                  <h5 className="text-normal margin-top-0 margin-bottom-2">
+                    {translateOperationalSolutionKey(solution.key)}
+                  </h5>
+                </>
+              )}
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ITSolutions/_components/CheckboxCard/index.tsx:179]: This conditional doesn't all for a rendering of both a default solution and a custom solution, which is what `Contractor` is for `Helpdesk support`

**Review by patrickseguraoddball — DISMISSED** [review]
The `otherHeader` isn't being shown here after I populate it in AddCustom.

Screenshot 2023-05-08 at 4 40 45 PM

**Review by patrickseguraoddball — APPROVED** [review]

**Pull request merged**